src/Cli/Microsoft.TemplateEngine.Cli/Commands/NewCommand.Legacy.cs

```diff
@@ -136,6 +136,7 @@ private static void ValidateArgumentUsage(CommandResult commandResult, params Cl
         private void BuildLegacySymbols(Func<ParseResult, ITemplateEngineHost> hostBuilder)
         {
             this.Arguments.Add(ShortNameArgument);
+            this.Arguments.Add(NameArgument);
```

Comment:
I'm not 100% certain how the legacy stuff works but this method is only called in the normal `NewCommand` constructor. There's nothing that signifies these as 'legacy' other than it being in this file. Since this `NameArgument` needed to be in-between the 2 other arguments, I didn't see a way to workaround editing this file. S.CL doesn't offer a way to rearrange arguments; they just get tossed into a `List` in the order which you call `Add`. There's no Remove method either. So, I'm a bit perplexed on this one.

Reply:
I see, i would rather see 138-140 in constructor instead before calling `BuildLegacySymbols` as those are _not_ legacy symbols, but it seems unrelated to the PR. So, that's up to you whether to fix it now or later.

Reply:
I moved these 3 arguments to the regular NewCommand constructor.